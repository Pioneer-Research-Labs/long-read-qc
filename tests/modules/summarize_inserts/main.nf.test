nextflow_process {

    name "Test Process summarize_inserts"
    script "modules/summarize_inserts/main.nf"
    process "summarize_inserts"

    test("Should generate insert summaries and plots") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = file('${projectDir}/test_data/insert_map.tsv')
                """
            }
        }

        then {
            assert process.success
            assert process.out.size() == 3
            assert process.out[0].size() == 1
            assert process.out[1].size() == 1
            assert process.out[2].size() == 1
            assert snapshot(
                "$outputDir/concatenated_inserts.csv",
                "$outputDir/insert_length_distribution.csv").match()
        }

    }

}
