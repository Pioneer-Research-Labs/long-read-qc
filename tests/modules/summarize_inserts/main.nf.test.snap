{
    "Should generate insert summaries and plots": {
        "content": [
            "concatenated_inserts.csv:md5,2656e79c1b1f8a54d5b1a5bf3f62d973",
            "insert_length_distribution.csv:md5,2656e79c1b1f8a54d5b1a5bf3f62d973"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-15T14:37:28.812598"
    }
}